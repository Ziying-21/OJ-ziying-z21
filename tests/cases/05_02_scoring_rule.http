###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user1"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:26 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user2"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:26 GMT
# 
# {"id":2,"name":"user2"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":2}
###
# Got response:
# HTTP 200
# content-length: 1137
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:26 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0},{"id":2,"info":"","memory":0,"result":"Waiting","time":0},{"id":3,"info":"","memory":0,"result":"Waiting","time":0},{"id":4,"info":"","memory":0,"result":"Waiting","time":0},{"id":5,"info":"","memory":0,"result":"Waiting","time":0},{"id":6,"info":"","memory":0,"result":"Waiting","time":0},{"id":7,"info":"","memory":0,"result":"Waiting","time":0},{"id":8,"info":"","memory":0,"result":"Waiting","time":0},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:27.418Z","id":0,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":2},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1222
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:28 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":177835},{"id":1,"info":"","memory":0,"result":"Accepted","time":63913},{"id":2,"info":"","memory":0,"result":"Accepted","time":1827},{"id":3,"info":"","memory":0,"result":"Accepted","time":1530},{"id":4,"info":"","memory":0,"result":"Accepted","time":1739},{"id":5,"info":"","memory":0,"result":"Accepted","time":1629},{"id":6,"info":"","memory":0,"result":"Accepted","time":1529},{"id":7,"info":"","memory":0,"result":"Accepted","time":1502},{"id":8,"info":"","memory":0,"result":"Accepted","time":1819},{"id":9,"info":"","memory":0,"result":"Accepted","time":1838},{"id":10,"info":"","memory":0,"result":"Accepted","time":1941}],"created_time":"2022-09-09T02:19:27.418Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":2},"updated_time":"2022-09-09T02:19:27.695Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 1122
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:28 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0},{"id":2,"info":"","memory":0,"result":"Waiting","time":0},{"id":3,"info":"","memory":0,"result":"Waiting","time":0},{"id":4,"info":"","memory":0,"result":"Waiting","time":0},{"id":5,"info":"","memory":0,"result":"Waiting","time":0},{"id":6,"info":"","memory":0,"result":"Waiting","time":0},{"id":7,"info":"","memory":0,"result":"Waiting","time":0},{"id":8,"info":"","memory":0,"result":"Waiting","time":0},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:28.437Z","id":1,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/1 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1207
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:29 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":206814},{"id":1,"info":"","memory":0,"result":"Accepted","time":59732},{"id":2,"info":"","memory":0,"result":"Accepted","time":1961},{"id":3,"info":"","memory":0,"result":"Accepted","time":1566},{"id":4,"info":"","memory":0,"result":"Accepted","time":1708},{"id":5,"info":"","memory":0,"result":"Accepted","time":1778},{"id":6,"info":"","memory":0,"result":"Accepted","time":1787},{"id":7,"info":"","memory":0,"result":"Accepted","time":1494},{"id":8,"info":"","memory":0,"result":"Accepted","time":1457},{"id":9,"info":"","memory":0,"result":"Accepted","time":1673},{"id":10,"info":"","memory":0,"result":"Accepted","time":1632}],"created_time":"2022-09-09T02:19:28.437Z","id":1,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":"2022-09-09T02:19:28.734Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 377
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:29 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:29.453Z","id":2,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/2 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 426
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:30 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":120490},{"id":1,"info":"","memory":0,"result":"Accepted","time":58842}],"created_time":"2022-09-09T02:19:29.453Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":0},"updated_time":"2022-09-09T02:19:29.638Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 200
# content-length: 377
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:30 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:30.465Z","id":3,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/3 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 426
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:31 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":123429},{"id":1,"info":"","memory":0,"result":"Accepted","time":57056}],"created_time":"2022-09-09T02:19:30.465Z","id":3,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1},"updated_time":"2022-09-09T02:19:30.651Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 377
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:31 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:31.479Z","id":4,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/4 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 432
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:32 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":123331},{"id":1,"info":"","memory":0,"result":"Wrong Answer","time":62430}],"created_time":"2022-09-09T02:19:31.479Z","id":4,"result":"Wrong Answer","score":0.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":0},"updated_time":"2022-09-09T02:19:31.670Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":1}
###
# Got response:
# HTTP 200
# content-length: 377
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:32 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:32.493Z","id":5,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":1},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/5 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 432
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:33 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":123983},{"id":1,"info":"","memory":0,"result":"Wrong Answer","time":58698}],"created_time":"2022-09-09T02:19:32.493Z","id":5,"result":"Wrong Answer","score":0.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World.\"); }","user_id":1},"updated_time":"2022-09-09T02:19:32.683Z"}
###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist?scoring_rule=latest HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 199
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:33 GMT
# 
# [{"rank":1,"scores":[0.0,0.0,100.0],"user":{"id":0,"name":"root"}},{"rank":1,"scores":[0.0,100.0,0.0],"user":{"id":2,"name":"user2"}},{"rank":3,"scores":[0.0,0.0,0.0],"user":{"id":1,"name":"user1"}}]
###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist?scoring_rule=highest HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 203
# content-type: application/json
# access-control-expose-headers: content-type
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# date: Fri, 09 Sep 2022 02:19:33 GMT
# 
# [{"rank":1,"scores":[100.0,0.0,100.0],"user":{"id":0,"name":"root"}},{"rank":2,"scores":[100.0,0.0,0.0],"user":{"id":1,"name":"user1"}},{"rank":2,"scores":[0.0,100.0,0.0],"user":{"id":2,"name":"user2"}}]
