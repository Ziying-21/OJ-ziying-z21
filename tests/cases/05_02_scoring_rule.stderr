[2022-09-09T02:19:26Z INFO  actix_server::builder] Starting 8 workers
[2022-09-09T02:19:26Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2022-09-09T02:19:27Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000171
[2022-09-09T02:19:27Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000489
[2022-09-09T02:19:27Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.001143
[2022-09-09T02:19:27Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1137 "-" "-" 0.000593
[2022-09-09T02:19:28Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 1222 "-" "-" 0.000368
[2022-09-09T02:19:28Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1122 "-" "-" 0.001193
[2022-09-09T02:19:29Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/1 HTTP/1.1" 200 1207 "-" "-" 0.000322
[2022-09-09T02:19:29Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 377 "-" "-" 0.001191
[2022-09-09T02:19:30Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/2 HTTP/1.1" 200 426 "-" "-" 0.000219
[2022-09-09T02:19:30Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 377 "-" "-" 0.000924
[2022-09-09T02:19:31Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/3 HTTP/1.1" 200 426 "-" "-" 0.000336
[2022-09-09T02:19:31Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 377 "-" "-" 0.001377
[2022-09-09T02:19:32Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/4 HTTP/1.1" 200 432 "-" "-" 0.000349
[2022-09-09T02:19:32Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 377 "-" "-" 0.001342
[2022-09-09T02:19:33Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/5 HTTP/1.1" 200 432 "-" "-" 0.000344
[2022-09-09T02:19:33Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist?scoring_rule=latest HTTP/1.1" 200 199 "-" "-" 0.000262
[2022-09-09T02:19:33Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist?scoring_rule=highest HTTP/1.1" 200 203 "-" "-" 0.000155
